src/sources/syslog.rs

```diff
@@ -215,10 +216,19 @@ fn event_from_str(host_key: &str, default_host: Option<Bytes>, line: &str) -> Op
         .as_mut_log()
         .insert(event::log_schema().source_type_key(), "syslog");
 
-    if let Some(host) = &parsed.hostname {
-        event.as_mut_log().insert(host_key, host.clone());
-    } else if let Some(default_host) = default_host {
-        event.as_mut_log().insert(host_key, default_host);
+    match (parsed.hostname, default_host) {
+        (Some(parsed_hostname), Some(default_host)) => {
+            event.as_mut_log().insert(host_key, parsed_hostname.clone());
+            event.as_mut_log().insert("source_ip", default_host);
+        },
+        (Some(parsed_hostname), None) => {
+            event.as_mut_log().insert(host_key, parsed_hostname.clone());
+        },
+        (None, Some(default_host)) => {
+            event.as_mut_log().insert(host_key, default_host.clone());
+            event.as_mut_log().insert("source_ip", default_host);
+        },
+        (None, None) => (),
     }
```

Comment:
Could this be something like the following?

```rust
if let Some(default_host) = default_host {
    event.as_mut_log().insert("source_ip", default_host);
}

if let Some(parsed_host) = parsed.hostname.or(default_host) {
    event.as_mut_log().insert(host_key, default_host.clone());
}
```

The combined match implies to me that there's some dependency between the two set fields, but I don't think there is.